src/components/TradeWidget/TxMessage.tsx

```diff
@@ -1,21 +1,94 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import { formatTimeInHours } from 'utils'
 import { useFormContext } from 'react-hook-form'
 import { TokenDetails } from 'types'
 import styled from 'styled-components'
 import { TradeFormData } from '.'
 import { displayTokenSymbolOrLink } from 'utils/display'
+import { usePriceEstimationInOwl, useWETHPriceInOwl } from 'hooks/usePriceEstimation'
+import BigNumber from 'bignumber.js'
+import { ZERO_BIG_NUMBER } from 'const'
+
+import alertIcon from 'assets/img/alert.svg'
+import { useGasPrice } from 'hooks/useGasPrice'
+import { DEFAULT_GAS_PRICE, calcMinTradableAmountInOwl } from 'utils/minFee'
 
 interface TxMessageProps {
   sellToken: TokenDetails
   receiveToken: TokenDetails
+  networkId: number
 }
 
 const TxMessageWrapper = styled.div`
   padding: 1em;
 `
 
-export const TxMessage: React.FC<TxMessageProps> = ({ sellToken, receiveToken }) => {
+const Warning = styled.p`
+  background: beige;
+  padding: 0.5em;
+  box-shadow: inset 0 0 7px 3px #cc262647;
+  border-radius: 0.3em;
+  display: flex;
+
+  .alert {
+    min-width: 2.5em;
+  }
+`
+
+interface LowVolumeParams {
+  sellToken: TokenDetails
+  networkId: number
+  sellTokenAmount: string
+}
+
+interface LowVolumeResult {
+  isLoading: boolean
+  isLowVolume?: boolean
+  difference?: BigNumber
+  minAmount?: BigNumber
+}
+
+const useLowVolumeAmount = ({ sellToken, sellTokenAmount, networkId }: LowVolumeParams): LowVolumeResult => {
+  const { priceEstimation, isPriceLoading } = usePriceEstimationInOwl({
+    tokenId: sellToken.id,
+    tokenDecimals: sellToken.decimals,
+    networkId,
+  })
+
+  const { priceEstimation: wethPriceInOwl, isPriceLoading: isWETHPriceLoading } = useWETHPriceInOwl(networkId)
+
+  const gasPrice = useGasPrice(DEFAULT_GAS_PRICE)
+
+  return useMemo(() => {
+    if (
+      isPriceLoading ||
+      isWETHPriceLoading ||
+      priceEstimation === null ||
+      wethPriceInOwl === null ||
+      gasPrice === null
+    )
+      return { isLoading: true }
+    console.log('priceEstimation of', sellToken.symbol, 'in OWL', priceEstimation.toString(10))
```

Comment:
please wrap in braces, very easy to miss if has a single statement

```suggestion
    if (
      isPriceLoading ||
      isWETHPriceLoading ||
      priceEstimation === null ||
      wethPriceInOwl === null ||
      gasPrice === null
    ) {
      return { isLoading: true }
    }

    console.log('priceEstimation of', sellToken.symbol, 'in OWL', priceEstimation.toString(10))
```